File: 백준/Silver/16112. 5차 전직/5차 전직.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

val br = BufferedReader(InputStreamReader(System.`in`))
val bw = System.out.bufferedWriter()

fun main() {
    val (n, k) = br.readLine().split(' ').map { it.toInt() }
    val input = StringTokenizer(br.readLine())
    val quest = MutableList(n){0L}
    var total = 0L

    for(i in 0 until n)
        quest[i] = input.nextToken().toLong()

    Collections.sort(quest)

    for(i in 0 until k)
        total += quest[i] * i

    for(i in k until n)
        total += quest[i] * k

    bw.write("$total")

    bw.close()
    br.close()
}

File: 백준/Silver/16112. 5차 전직/5차 전직.py
import sys

n, k = map(int, sys.stdin.readline().split())
quest = list(map(int, sys.stdin.readline().split()))
quest.sort()
total = 0

for i in range(1, n):
    if i < k:
        total += quest[i] * i
    else:
        total += quest[i] * k

print(total)
